gcloud-java-compute/src/test/java/com/google/gcloud/compute/SnapshotTest.java

```diff
@@ -0,0 +1,253 @@
+    assertEquals(LICENSES, expectedSnapshot.licenses());
+    assertEquals(SOURCE_DISK, expectedSnapshot.sourceDisk());
+    assertEquals(SOURCE_DISK_ID, expectedSnapshot.sourceDiskId());
+    assertEquals(STORAGE_BYTES, expectedSnapshot.storageBytes());
+    assertEquals(STORAGE_BYTES_STATUS, expectedSnapshot.storageBytesStatus());
+    assertSame(serviceMockReturnsOptions, expectedSnapshot.compute());
+    SnapshotId otherSnapshotId = SnapshotId.of("otherSnapshot");
+    DiskId otherSourceDisk = DiskId.of("zone", "otherDisk");
+    Snapshot snapshot = new Snapshot.Builder(serviceMockReturnsOptions, SNAPSHOT_ID, SOURCE_DISK)
+        .snapshotId(otherSnapshotId)
+        .sourceDisk(otherSourceDisk)
+        .build();
+    assertNull(snapshot.id());
+    assertEquals(otherSnapshotId, snapshot.snapshotId());
+    assertNull(snapshot.creationTimestamp());
+    assertNull(snapshot.description());
+    assertNull(snapshot.status());
+    assertNull(snapshot.diskSizeGb());
+    assertNull(snapshot.licenses());
+    assertEquals(otherSourceDisk, snapshot.sourceDisk());
+    assertNull(snapshot.sourceDiskId());
+    assertNull(snapshot.storageBytes());
+    assertNull(snapshot.storageBytesStatus());
+    assertSame(serviceMockReturnsOptions, snapshot.compute());
+  }
+
+  @Test
+  public void testToAndFromPb() {
+    initializeExpectedSnapshot(8);
+    compareSnapshot(expectedSnapshot,
+        Snapshot.fromPb(serviceMockReturnsOptions, expectedSnapshot.toPb()));
+    Snapshot snapshot =
+        new Snapshot.Builder(serviceMockReturnsOptions, SNAPSHOT_ID, SOURCE_DISK).build();
+    compareSnapshot(snapshot, Snapshot.fromPb(serviceMockReturnsOptions, snapshot.toPb()));
+  }
+
+  @Test
+  public void testDeleteOperation() {
+    initializeExpectedSnapshot(2);
+    expect(compute.options()).andReturn(mockOptions);
+    Operation operation = new Operation.Builder(serviceMockReturnsOptions)
+        .operationId(GlobalOperationId.of("project", "op"))
+        .build();
+    expect(compute.deleteSnapshot(SNAPSHOT_ID)).andReturn(operation);
+    replay(compute);
+    initializeSnapshot();
+    assertSame(operation, snapshot.delete());
+  }
+
+  @Test
+  public void testDeleteNull() {
+    initializeExpectedSnapshot(1);
+    expect(compute.options()).andReturn(mockOptions);
+    expect(compute.deleteSnapshot(SNAPSHOT_ID)).andReturn(null);
+    replay(compute);
+    initializeSnapshot();
+    assertNull(snapshot.delete());
+  }
+
+  @Test
+  public void testExists_True() throws Exception {
+    initializeExpectedSnapshot(1);
+    Compute.SnapshotOption[] expectedOptions = {Compute.SnapshotOption.fields()};
+    expect(compute.options()).andReturn(mockOptions);
+    expect(compute.getSnapshot(SNAPSHOT_ID.snapshot(), expectedOptions))
+        .andReturn(expectedSnapshot);
+    replay(compute);
+    initializeSnapshot();
+    assertTrue(snapshot.exists());
+    verify(compute);
+  }
+
+  @Test
+  public void testExists_False() throws Exception {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Method names should not contain underscores](https://www.codacy.com/app/mziccard/gcloud-java/file/2311979398/issues/source?bid=3140968&fileBranchId=3194371#l190)